src/modules/chapter/components/Chapter.tsx

```diff
@@ -30,14 +37,17 @@ const Chapter = () => {
             useMangaStore.setState({ selectedChapter: oldChapterState })
             return (
                 useMangaStore.getState().selectedChapter &&
-
-
-                <div className="flex flex-col gap-4">
-                    <ChapterNavBar chapterName={useMangaStore.getState().selectedChapter!.title} chapterNumber={useMangaStore.getState().selectedChapter!.chapterNumber} mangaName={useMangaStore.getState().selectedManga!.title} />
-                    <h1 className="place-self-center text-2xl">{useMangaStore.getState().selectedManga!.title}</h1>
-                    <h2 className="place-self-center mb-12">{useMangaStore.getState().selectedChapter?.chapterNumber}. {useMangaStore.getState().selectedChapter!.title}</h2>
-                    <Pages pages={useMangaStore.getState().selectedChapter!.pages!} />
-                </div>
+                <>
+                    <motion.div initial="hidden" whileInView="visible" style={{ scaleX }} className="fixed top-0 left-0 right-0 bg-rose-400 origin-left h-2 z-10" viewport={{ root: pagesRef }} />
+                    <div className="flex flex-col gap-4" ref={pagesRef}>
+                        <ChapterNavBar chapterName={useMangaStore.getState().selectedChapter!.title} chapterNumber={useMangaStore.getState().selectedChapter!.chapterNumber} mangaName={useMangaStore.getState().selectedManga!.title} />
```

Comment:
Maybe try 
```
{useMangaStore.getState().selectedChapter?.title  &&  <ChapterNavBar chapterName={useMangaStore.getState().selectedChapter.title} ... }
```
To remove those assertions

Reply:
It's not working due to how the type is within my store. I put null as an option so I can give it the initial value of null instead of an empty object